lib/awsutil/credentials.go

```diff
@@ -52,7 +66,8 @@ func tryLoadCredentialsWithOptions(
 	if err != nil {
 		return nil, nil, err
 	}
-	cs, unloadableAccounts := createCredentials(accountNames, options)
+	cs, unloadableAccounts := createCredentials(
+		noDefault(accountNames), options)
```

Comment:
I would prefer to see this filtering out of `default` done in `listAccountNames()`.
Also, it would be nice if `loadCredentials()` is a thin wrapper around `tryLoadCredentialsWithOptions()`.